line_profiler/_line_profiler.pyx

```diff
@@ -339,10 +339,23 @@ cdef class LineProfiler:
                 entries += list(cmap[entry].values())
             key = label(code)
 
-            entries = [(e["lineno"], e["nhits"], e["total_time"]) for e in entries]
+            temp_entries = [(e["lineno"], e["nhits"], e["total_time"]) for e in entries]
             # If there are multiple entries for a line, this will sort them by increasing # hits
-            entries.sort()
-
+            temp_entries.sort()
+            # If a line causes a branch, the stats for the True branch and False branch are separate.
+            # So we merge them here with 2 passes
+            branch_entries = {}
+            # Find the indicies of items with the same line number
+            for indx, entry in enumerate(temp_entries):
+                branch_entries[entry[0]] = branch_entries.setdefault(entry[0], []) + [indx]
+            entries = []
+            for entry in branch_entries:
```

Comment:
This entire index-storing runaround seems excessive to me, I think something like this immediately doing the sum work might be better:
```
...
entries.sort()

# Merge duplicate line numbers, which occur for branch entrypoints like `if`
nhits_by_lineno = {}
total_time_by_lineno = {}
for lineno, nhits, total_time in entries:
     nhits_by_lineno[lineno] = nhits_by_lineno.setdefault(lineno, 0) + nhits
     total_time_by_lineno[lineno] = nhits_by_lineno.setdefault(lineno, 0) + total_hits
entries = [(lineno, nhits, total_time_by_lineno[lineno]) for lineno, nhits in nhits_by_lineno.items()]
```

Probably some non-compiling error in the above but you get the gist

Reply:
Though it occurs to me now that this code might need to work in python versions before dicts were ordered, in which case this doesn't work

Reply:
Looks like that could work, I'll test it when I can!